base/src/main/java/com/smartdevicelink/managers/screen/choiceset/PreloadPresentChoicesOperation.java

```diff
@@ -0,0 +1,820 @@
+        this.currentState = SDLPreloadPresentChoicesOperationState.UPLOADING_IMAGES;
+
+        List<SdlArtwork> artworksToUpload = artworksToUpload();
+
+        if (artworksToUpload.size() == 0) {
+            DebugTool.logInfo(TAG, "Choice Preload: No Choice Artworks to upload");
+            listener.onComplete(true);
+            return;
+        }
+
+        if (fileManager.get() != null) {
+            fileManager.get().uploadArtworks(artworksToUpload, new MultipleFileCompletionListener() {
+                @Override
+                public void onComplete(Map<String, String> errors) {
+                    if (errors != null && errors.size() > 0) {
+                        DebugTool.logError(TAG, "Error uploading choice cell Artworks: " + errors.toString());
+                        listener.onComplete(false);
+                    } else {
+                        DebugTool.logInfo(TAG, "Choice Artworks Uploaded");
+                        listener.onComplete(true);
+                    }
+                }
+            });
+        } else {
+            DebugTool.logError(TAG, "File manager is null in choice preload operation");
+            listener.onComplete(false);
+        }
+    }
+
+    private void preloadCells(@NonNull final CompletionListener listener) {
+        this.currentState = SDLPreloadPresentChoicesOperationState.UPLOADING_CHOICES;
+
+        List<CreateInteractionChoiceSet> choiceRPCs = new ArrayList<>(cellsToUpload.size());
+        for (ChoiceCell cell : cellsToUpload) {
+            CreateInteractionChoiceSet csCell = choiceFromCell(cell);
+            if (csCell != null) {
+                choiceRPCs.add(csCell);
+            }
+        }
+
+        if (choiceRPCs.size() == 0) {
+            DebugTool.logError(TAG, " All Choice cells to send are null, so the choice set will not be shown");
+            listener.onComplete(true);
+            return;
+        }
+
+        if (internalInterface.get() != null) {
+            internalInterface.get().sendRPCs(choiceRPCs, new OnMultipleRequestListener() {
+                @Override
+                public void onUpdate(int remainingRequests) {
+
+                }
+
+                @Override
+                public void onFinished() {
+                    DebugTool.logInfo(TAG, "Finished pre loading choice cells");
+                    listener.onComplete(!choiceError);
+                    choiceError = false;
+                }
+
+                @Override
+                public void onResponse(int correlationId, RPCResponse response) {
+                    if (!response.getSuccess()) {
+                        DebugTool.logError(TAG, "There was an error uploading a choice cell: " + response.getInfo() + " resultCode: " + response.getResultCode());
+                        choiceError = true;
+                    } else {
+                        loadedCells.add(cellFromChoiceId(correlationId));
```

Comment:
correlationID does not translate to choiceCell ID so loadedCells are always empty, prompting cells to be uploaded everytime you go to present